src/main/kotlin/com/defenseunicorns/koci/Layout.kt

```diff
@@ -192,6 +192,10 @@ class Layout private constructor(
     @OptIn(ExperimentalSerializationApi::class)
     @Suppress("detekt:LongMethod", "detekt:CyclomaticComplexMethod")
     suspend fun remove(descriptor: Descriptor): Result<Boolean> = runCatching {
```

Comment:
remove should lock the mutex as well - a download could try to start during a removal

Reply:
i placed a guard for this, but downloads only currently track at the layer level, there is no "lock" for a manifest/index

trying to think of the scenario where this could cause a bad state to occur

Reply:
The main issue is that this implementation relies on filesystem implementations of file descriptors/handles to handle concurrency between `push` and `remove` which could result in different behavior on different systems depending on how they implemented that functionality - because a `file.delete()` and a `FileOutputStream().use` could be called simultaneously (since the guard only guards `remove`s entrypoint) this would fallback onto that machine's implementation for deconfliction.  A more robust approach would mutex both ends of that.